mne/io/eeglab/eeglab.py

```diff
@@ -23,16 +23,33 @@
 CAL = 1e-6
 
 
-def _check_fname(fname):
-    """Check if the file extension is valid."""
-    fmt = str(op.splitext(fname)[-1])
+def _check_fname(fname, dataname):
+    """Check whether the filename is valid.
+
+    Check if the file extension is ``.fdt`` (older ``.dat`` being invalid) or
+    whether the ``EEG.data`` filename exists. If ``EEG.data`` file is absent
+    the set file name with .set changed to .fdt is checked.
+    """
+    fmt = str(op.splitext(dataname)[-1])
     if fmt == '.dat':
         raise NotImplementedError(
             'Old data format .dat detected. Please update your EEGLAB '
             'version and resave the data in .fdt format')
     elif fmt != '.fdt':
         raise IOError('Expected .fdt file format. Found %s format' % fmt)
 
+    basedir = op.dirname(fname)
+    data_fname = op.join(basedir, dataname)
+    if not op.exists(data_fname):
+        fdt_from_set_fname = op.splitext(fname)[0] + '.fdt'
+        if op.exists(fdt_from_set_fname):
+            data_fname = fdt_from_set_fname
+            msg = ('Data filename in EEG.data ({}) is incorrect, using the '
+                   'correct file name ({}).')
+            warn(msg.format(dataname, op.basename(fdt_from_set_fname)))
+        # else raise error?
```

Comment:
I would say yes. This was not captured by a test? if there is not file containing the data it should crash no?

Reply:
Yes, if the file is missing there would be an error later on, the only benefit here would be that the error could say that files corresponding to both file names are absent.